de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/NewFolderHandler.java

```diff
@@ -1,50 +1,34 @@
- 
+
 package de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers;
 
-import java.nio.file.Path;
+import java.io.File;
 
 import org.eclipse.e4.core.di.annotations.CanExecute;
 import org.eclipse.e4.core.di.annotations.Execute;
-import org.eclipse.swt.graphics.Point;
-import org.eclipse.swt.widgets.Shell;
 
 import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
-import de.tu_bs.cs.isf.e4cf.core.file_structure.WorkspaceFileSystem;
-import de.tu_bs.cs.isf.e4cf.core.file_structure.util.FileHandlingUtility;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPDialogService;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
 import de.tu_bs.cs.isf.e4cf.core.util.services.RCPSelectionService;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.NewFolderPage;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.NewFolderWizard;
 
 public class NewFolderHandler {
-	
+
+	private static final String FOLDER_PLACEHOLDER = "/New Folder";
+
 	@Execute
-	public void execute(Shell shell, RCPDialogService dialogService, RCPSelectionService selectionService, 
-			RCPImageService imageService, WorkspaceFileSystem fileSystem) {
-		FileTreeElement selectedElement = selectionService.getCurrentSelectionFromExplorer();
-		
-		NewFolderPage page = new NewFolderPage(
-				"NewFolder", "Choose a Folder Name", 
-				imageService.getImageDescriptor(null, "icons/Explorer_View/items/folder24.png")
-		);
-		
-		// create selected directory when selected element available, else create new project
-		Path directory;
-		if (selectedElement != null) {
-			directory = FileHandlingUtility.getPath(selectedElement);
-		} else {
-			directory = FileHandlingUtility.getPath(fileSystem.getWorkspaceDirectory());
+	public void execute(RCPSelectionService rcpSelectionService) {
+		FileTreeElement element = rcpSelectionService.getCurrentSelectionFromExplorer();
+		String stringPath = element.getAbsolutePath();
+		File dir = new File(stringPath + FOLDER_PLACEHOLDER);
+		int i = 2;
```

Comment:
"i" is not a speaking variable name as it is not exclusively used as loop-variable here

Reply:
renamed